src/query-compiler/default-query-compiler.ts

```diff
@@ -1509,6 +1531,14 @@ export class DefaultQueryCompiler
     }
   }
 
+  protected override visitTop(node: TopNode): void {
+    this.append(`top(${node.expression})`)
```

Comment:
There's an injection vulnerability for both `expression` and `modifiers`. I could imagine client code that takes the expression and/or modifiers straight from user input.

Add a "parser" that checks that `expression` and `modifier` are in fact a number and the literal `percent` before creating the node.

Reply:
Great catch!

done.